cfme/tests/middleware/test_middleware_utilization.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+from cfme.middleware.datasource import MiddlewareDatasource
+from cfme.middleware.messaging import MiddlewareMessaging
+from cfme.middleware.server import MiddlewareServer
+from cfme.middleware.utilization import Option
+from random_methods import get_random_object
+from utils import testgen
+from utils.version import current_version
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.uncollectif(lambda: current_version() < '5.7'),
+]
+pytest_generate_tests = testgen.generate(testgen.provider_by_type, ["hawkular"], scope="function")
+
+
+@pytest.mark.parametrize("object_type", [MiddlewareServer,
+                                         MiddlewareDatasource,
+                                         MiddlewareMessaging])
+def test_object_utilization(provider, object_type):
+    """Tests utilization charts in all pages that has utilization charts
+
+    Steps:
+        * Select a utilization object of provided object_type randomly from database
+        * Run `validate_utilization` with `utilization_obj` input
+    """
+    utilization_obj = get_random_object(provider=provider, objecttype=object_type, load_from="db")
+    validate_utilization(utilization_obj=utilization_obj.utilization)
+
+
+def validate_utilization(utilization_obj):
+    """Gets data from chart, table, mgmt and compare between"""
+    for chart_id in utilization_obj.charts:
+        chart = getattr(utilization_obj, chart_id)
+        # validate legends
+        validate_legends(chart=chart)
+        # validate data for Daily with 1 Week
```

Comment:
Can you group and iterate these. Looks like you have at least 2 loops where op_interval or op_week is static and the other is being varied.

```python
for op_week in [WK_1_WEEK, WK_2_WEEK, WK_3_WEEK, WK_4_WEEK]
    chart.option.set_by_visible_text(op_interval=Option.IN_DAILY, op_week=Option.op_week)
    validate_data(chart=chart)

No need to have this expanded out, I don't think it aids in readability or clarity in what is being tested. You could even take it a step further and iterate over op_interval in [IN_DAILY, IN_MOST_RECENT_HOUR] to express all of this in 5-6 lines.